woocommerce/woocommerce-android
# Hilt migration: migrate AppComponent and Application class injection

First part of #3936, this migrates the app modules to Hilt, and migrates the application's injection too.

I know this is a big PR, but I wanted to fix the UI tests in the same PR, instead of having them in a separate PR, as they are related, and a reason of the change to the app initialization logic.

#### Testing
1. Make the app compiles, and perform some tasks to confirm that activities and fragments are correctly injected.
2. Run the UI tests and confirm that they will reach the password screen before failing, the failure after that is not related to this PR, as they weren't updated for the unified login process.

Update release notes:
